SFR/Pincell/make_openmc_model.py

```diff
@@ -1,109 +1,69 @@
 
     return ap.parse_args()
 
 
-def model_generate(arguments):
-    """
-    :return:
-    a pincell universe,
-    openmc.Materials class,
-    openmc.Geometry class,
-    openmc.Settings class
-
-    the universe class is mostly for reuse if we want to create an assembly
-    """
+def get_pincell_universe(arguments):
     fuel_mat_name = f'{arguments.pincell_type}_fuel'
+
     fuel_material = make_sfr_material(material_dict[fuel_mat_name], percent_type='wo')
     cladding_material = make_sfr_material(material_dict['cladding'], percent_type='ao')
-    sodium = make_sfr_material(material_dict['sodium'], percent_type='ao')
     helium = make_sfr_material(material_dict['helium'], percent_type='ao')
+    sodium = make_sfr_material(material_dict['sodium'], percent_type='ao')
 
     fuel_or = openmc.ZCylinder(r=pincell_params.r_fuel)
     clad_ir = openmc.ZCylinder(r=pincell_params.r_clad_inner)
     clad_or = openmc.ZCylinder(r=(pincell_params.r_clad_inner + pincell_params.t_clad))
-    fuel_bb = openmc.model.RectangularPrism(width=pincell_params.pitch,
-                                            height=pincell_params.height / arguments.n_axial,
-                                            boundary_type="reflective")
-    top = openmc.ZPlane(z0=pincell_params.height, boundary_type="reflective")
-    bottom = openmc.ZPlane(z0=0, boundary_type="vacuum")
 
     cladding_cell = openmc.Cell(fill=cladding_material, region=+clad_ir & -clad_or)
     gas_gap_cell = openmc.Cell(fill=helium, region=+fuel_or & -clad_ir)
     fuel_cell = openmc.Cell(fill=fuel_material, region=-fuel_or)
-    sodium_cell = openmc.Cell(region=+clad_or & -fuel_bb & +bottom & - top, fill=sodium)
-    pincell_universe = openmc.Universe(cells=[cladding_cell, gas_gap_cell, fuel_cell])
-
-    pincell_lattice = openmc.RectLattice()
-    pincell_lattice.pitch = (
-        pincell_params.pitch, pincell_params.pitch,
-        pincell_params.height / arguments.n_axial)
-    pincell_lattice.lower_left = (
-        -pincell_params.pitch / 2.0, -pincell_params.pitch / 2.0, 0.0)
+    sodium_cell = openmc.Cell(fill=sodium, region=+clad_or)
+
+    return openmc.Universe(cells=[cladding_cell, gas_gap_cell, fuel_cell, sodium_cell]), openmc.Materials([fuel_material, helium, cladding_material, sodium])
+
+
+def generate_pincell_model(arguments):
+    pincell_universe, material = get_pincell_universe(arguments)
+
+    fuel_bb = openmc.model.HexagonalPrism(edge_length=pincell_params.edge_length, boundary_type="reflective", orientation='y')
+    top = openmc.ZPlane(z0=pincell_params.height, boundary_type="reflective")
+    bottom = openmc.ZPlane(z0=0, boundary_type="reflective")
+
+    sodium = make_sfr_material(material_dict['sodium'], percent_type='ao')
+    material.append(sodium)
+
+    pincell_lattice = openmc.HexLattice()
+    pincell_lattice.center = (0.0, 0.0, 0.0)
+    pincell_lattice.orientation = "y"
+    pincell_lattice.outer = openmc.Universe(cells=[openmc.Cell(fill=sodium)])
```

Comment:
The LWR version does not need to define the `outer` and therefore doesn't need to make a universe like this.

Reply:
Just for my own personal curiosity, why does the pincell model not use the `lattice.outer`? Does it explicitly define the universe outside of a pin? For the cnerg/gcmr HexLattice, I do take advantage of something similar to line 58.

Reply:
It defines the space outside of the pin as infinitely large and filled with coolant

Reply:
It may be redundant in your models, too, since you define the space outside your compacts as part of each compact universe.